### PR TITLE
feat: [CI-4828]: add hosted option to multiple connectors

### PR DESCRIPTION
**Pull request by d1wilko**
### Summary

Add hosted option to the following connectors

- Docker
- Artifactory
- GCP
- AWS
- Azure

#### Screenshots

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 15 Issues 
* Code issue 15 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10579)

**Comment by d1wilko** [comment]
retrigger titlecheck

**Comment by d1wilko** [comment]
sonarqube is failing a security check based on a test mock password - ignoring

File: src/modules/35-connectors/components/CreateConnector/AWSConnector/__test__/CreateAWSConnector.test.tsx [hunk 1]
```diff
@@ -115,6 +128,56 @@ describe('Create AWS connector Wizard', () => {
     )
   })
 
+  test('Should render form for edit for hosted', async () => {
+    const { container } = render(
+      <TestWrapper path="/account/:accountId/resources/connectors" pathParams={{ accountId: 'dummy' }}>
```
Comment (reetika04) [inline src/modules/35-connectors/components/CreateConnector/AWSConnector/__test__/CreateAWSConnector.test.tsx:133]: Please use routes.

File: src/modules/35-connectors/components/CreateConnector/AWSConnector/__test__/CreateAWSConnector.test.tsx [hunk 2]
```diff
@@ -115,6 +128,56 @@ describe('Create AWS connector Wizard', () => {
...
+          {...commonProps}
+          isEditMode={true}
+          connectorInfo={hostedMockConnector.data.connector as any}
+          mock={mockResponse}
+          connectivityMode={ConnectivityModeType.Manager}
+        />
+      </TestWrapper>
+    )
+    // editing connector name
+    const updatedName = 'dummy name'
```
Comment (reetika04) [inline src/modules/35-connectors/components/CreateConnector/AWSConnector/__test__/CreateAWSConnector.test.tsx:144]: Updating to the same name?

**Review by reetika04 — APPROVED** [review]

**Review by mayankVermaHarness — DISMISSED** [review]
Please check Reetika's comments.

File: src/modules/35-connectors/components/CreateConnector/AWSConnector/__test__/CreateAWSConnector.test.tsx (hunk 2 above)
Comment (d1wilko) [reply src/modules/35-connectors/components/CreateConnector/AWSConnector/__test__/CreateAWSConnector.test.tsx:144]: no it is initially `AWS test` from the mock - but I'll make a small change to make that clearer :)

File: src/modules/35-connectors/components/CreateConnector/AWSConnector/__test__/CreateAWSConnector.test.tsx (hunk 1 above)
Comment (d1wilko) [reply src/modules/35-connectors/components/CreateConnector/AWSConnector/__test__/CreateAWSConnector.test.tsx:133]: will do

**Review by reetika04 — DISMISSED** [review]

**Pull request merged**